hazelcast/src/main/java/com/hazelcast/map/impl/nearcache/invalidation/BatchInvalidator.java

```diff
@@ -110,40 +119,30 @@ private void invalidate(InvalidationQueue invalidationQueue, boolean offloadEven
             return;
         }
 
-        List<SingleNearCacheInvalidation> invalidations = createInvalidations(invalidationQueue, size);
+        List<Invalidation> invalidations = createInvalidations(invalidationQueue, size);
         try {
             sendInvalidations(invalidations, offloadEventSending, mapName);
         } finally {
             invalidationQueue.release();
         }
     }
 
-    private static List<SingleNearCacheInvalidation> createInvalidations(InvalidationQueue invalidationQueue, int size) {
-        boolean foundClearEvent = false;
-        List<SingleNearCacheInvalidation> invalidations = new ArrayList<SingleNearCacheInvalidation>(size);
+    private static List<Invalidation> createInvalidations(InvalidationQueue invalidationQueue, int size) {
+        List<Invalidation> invalidations = new ArrayList<Invalidation>(size);
 
         for (int i = 0; i < size; i++) {
-            SingleNearCacheInvalidation invalidation = invalidationQueue.poll();
+            Invalidation invalidation = invalidationQueue.poll();
             if (invalidation == null) {
                 break;
             }
 
-            if (foundClearEvent) {
-                continue;
-            }
-
-            if (invalidation.getKey() == null) {
-                foundClearEvent = true;
-                invalidations = singletonList(invalidation);
-            } else {
-                invalidations.add(invalidation);
-            }
+            invalidations.add(invalidation);
```

Comment:
Shouldn't we keep the optimization to just send a singleton list if there is a `ClearNearCacheInvalidation`?
```
if (invalidation instanceof ClearNearCacheInvalidation) {
    return singletonList(invalidation);
}
```